deploy/config/deploy/yum.rb

```diff
@@ -63,4 +64,8 @@
       run "if [ -e #{repo_instance_path}-previous ]; then rm -rf #{repo_instance_path}-previous; fi" if overwrite || merge
     end
   end
+
+  task :purgecdn do
+    run "find #{repo_path} -mtime -3 | sed 's|#{deploy_to}|https://yum.theforeman.org/|' | xargs --no-run-if-empty curl -X PURGE -H 'Fastly-Soft-Purge:1'"
```

Comment:
one could also use something like `rsync … --log-file rsync.log --log-file-format "CHANGED %o %f"` and then `grep` out the `CHANGED` lines from the log and purge these names

Reply:
Looks like `%i` can tell whether it was actually changed. Is it safe enough to use `%i %f` and grep for c in the first column? Untested, but  `awk '$1 ~ /c/ { print $2 }' rsync.log`

Reply:
you line looks like this:

```
2019/12/04 10:13:22 [13933] >f+++++++++ sles12/x86_64/subscription-manager-1.23.3-4.1.suse1315.x86_64.rpm
```

and for comparison, mine:

```
2019/12/04 09:50:19 [13139] CHANGED recv sles12/x86_64/subscription-manager-1.23.3-4.1.suse1315.x86_64.rpm
```

Reply:
After some manual testing it looks like yours is good enough. It does look like `--log-file` appends so we should make sure it's clean before we start. Other than that I think the rsync approach is nice. Given the foreman-release handling is manual, you probably need to purge those explicitly.

Reply:
That's exactly what I thought too, so updated this PR to match that.